### PR TITLE
Dev containerservice microsoft.container service 2019 04 01

### PR DESCRIPTION
**Pull request by xizha162**
Merge 2019-04-01 container service from dev branch to master
The following changes are included and already approved when merging to the dev branch:

Adding windowsProfile and maxAgentPools property in the new 2019-04-01 (#5465)
adding load balancer sku property (#5556)
Add support VMSS agent nodes with public IP. (#5748)
Add identity property in managedCluster definition and add an example(#5709)
node resource group can be read-write(#5762)
Update managed cluster upgrades(#5789)

I also included a minor change to include 2019-04-01 in 2019-04 package.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6000 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6000/commits/ab580a24f4012bb4f8689b66003b2467091fdd43

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json [hunk 1]
```diff
@@ -0,0 +1,2087 @@
...
+          "description": "The principal id of the system assigned identity which is used by master components."
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant id of the system assigned identity which is used by master components."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of identity used for the managed cluster. Type 'SystemAssigned' will use an implicityly created identity in master components and an auto-created user assigned identity in MC_ resource group in agent nodes. Type 'None' will not use MSI for the managed cluster, service principal will be used instead.",
```
Comment (nschonni) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json]:
```suggestion
          "description": "The type of identity used for the managed cluster. Type 'SystemAssigned' will use an implicitly created identity in master components and an auto-created user assigned identity in MC_ resource group in agent nodes. Type 'None' will not use MSI for the managed cluster, service principal will be used instead.",
```

Comment (praries880) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json]: @xizhamsft Kindly take care of the suggestion here.. rest looks good to me.

Comment (xizhamsft) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json]: @praries880 , I just fixed the typo with a new commit.

**Review by praries880 — APPROVED** [review]

**Pull request merged**